### PR TITLE
Ci merge

### PR DESCRIPTION
**Pull request by arctan5x**
Introducing CircleCI for Docker-Splunk Repo

### CODE REVIEW

File: tests/test_output/1so_apps.json [hunk 1]
```diff
@@ -0,0 +1,109 @@
+{
```
Comment (nwang92) [inline tests/test_output/1so_apps.json]: I think 100% object-matching like this will cause a lot of build instability. If you make any changes in splunk-ansible, this will break. If we want to do exact matches like this, it should be done closer to the source of the code generating this blob.

On the other-hand, I get the intention for checking that the inventory + vars line up. For each of the scenarios, it's best to make sure certain keys have well-defined values (ex. in this case, make sure `json["all"]["vars"]["splunk"]["apps_location"]` is defined and has the value you expect) vs. doing a full comparison of everything.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,52 +1,65 @@
-# Python CircleCI 2.0 configuration file
-#
-# Check https://circleci.com/docs/2.0/language-python/ for more details
-#
 version: 2
 jobs:
-  build:
```
Comment (lephino) [inline .circleci/config.yml:7]: If we're going to "rebuild" for every test, it'd be faster to add 1 build step and use docker save to save the artifact and docker load on each of the test phases.

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,52 +1,65 @@
...
-    machine: true
-    #support this after we can enable it on account
-    #docker_layer_caching: true
+  test_centos7:
+    machine:
```
Comment (lephino) [inline .circleci/config.yml:4]: be warned, provisioning "machine" over and over actually allocates a full vm to us.  We pay for a plan based on the number of those that run.  We're good if people run this single threaded, but if parallelism gets added, this will really drive up our costs.

File: .circleci/config.yml [hunk 4]
```diff
@@ -1,52 +1,65 @@
...
+      image: circleci/classic:latest
+      docker_layer_caching: true
     working_directory: ~/repo
-
     steps:
-      # this section isn't needed for machine executors
-      #- setup_remote_docker
-           # enable this once the account supports it
-           #docker_layer_caching: true
```
Comment (lephino) [inline .circleci/config.yml:17]: I'll check and see if we're getting this in our plan, if so this would mean you don't need to do a "docker save / load" and can just reuse the build cache.

File: .gitignore [hunk 5]
```diff
@@ -8,3 +8,4 @@ splunk-ansible
 *.xml
 *clair-scanner*
 .env
+test-results/*
```
Comment (lephino) [inline .gitignore]: this is already dealt with by the !.gitignore in this directory.  They chain.

File: Makefile [hunk 6]
```diff
@@ -120,33 +119,27 @@ sample-compose-up: sample-compose-down
...
-	docker cp ${TEST_IMAGE_NAME}:$(shell pwd)/testresults.xml testresults.xml || echo "no testresults.xml"
+run_tests_centos7:
+	@echo 'Running the super awesome tests; CentOS 7'
+	pytest -sv tests/test_centos_7.py --junitxml test-results/centos7-result/testresults_centos7.xml
 
 test_setup:
 	@echo 'Install test requirements'
 	pip install -r $(shell pwd)/tests/requirements.txt --upgrade
+	mkdir test-results
 	mkdir test-results/pytest
```
Comment (lephino) [inline Makefile]: if you're making subdir's for centos / debian, do you need this pytest dir?

File: Makefile [hunk 7]
```diff
@@ -177,6 +170,6 @@ clean:
 	docker rm clair || true
 	rm -rf clair-scanner || true
 	rm -rf clair-scanner-logs || true
-	rm -rf test-results/* || true
+	rm -rf test-results || true
```
Comment (lephino) [inline Makefile]: Don't do this, it'll blow away the !.gitnore file for the subdir.  This dir should always be there, and we only want to remove the non-hidden contents of the dir.

File: test-results/.gitignore [hunk 8]
```diff
@@ -1,2 +0,0 @@
-*
```
Comment (lephino) [inline test-results/.gitignore]: why are we removing this?

File: Makefile (hunk 7 above)
Comment (gracening) [reply Makefile]: why do we need the test-results directory to persist? What's the point of having the empty directory in the repository?

Comment (lephino) [reply Makefile]: You don't know what order someone locally is going to run things, and what make targets they'll use.  If I for example run clair without tests, then the clair make point is required to create the entire directory structure.  The management of what needs to run setup and cleanup just becomes cumbersome to manage, vs, leaving a set location all items are expected to write to.  As tests / jobs / options expand, it provides a set location that people will look to and use without needing to be instructed, and prevents them from creating a divergence and making their own folders.

File: Makefile (hunk 6 above)
Comment (gracening) [reply Makefile]: nope lol

**Review by nwang92 — DISMISSED** [review]

**Review by bb03 — DISMISSED** [review]

**Review by nwang92 — DISMISSED** [review]

File: tests/test_debian_9.py [hunk 9]
```diff
@@ -409,66 +445,165 @@ def test_compose_1so_trial(self):
...
         # Wait for containers to be healthy
         assert self.wait_for_containers(container_count)
         # Check Splunkd on all the containers
         assert self.check_splunkd("admin", self.password)
+        # Check ansible version & configs
+        self.check_ansible(output)
+        # Check values in log output
+        self.check_common_keys(log_json, "so")
     
     def test_compose_1so_custombuild(self):
```
Comment (jmeixensperger) [inline tests/test_debian_9.py:460]: Can we can parametrize some of these tests to group them together? Some are very similar and it seems we could reduce duplicate code.

Comment (arctan5x) [reply tests/test_debian_9.py:460]: I see. Though these are testing different scenarios on compose so we don't want to merge the test cases.

**Review by nwang92 — APPROVED** [review]

**Review by gracening — APPROVED** [review]

**Review by bb03 — APPROVED** [review]

File: tests/test_debian_9.py (hunk 9 above)
Comment (jmeixensperger) [reply tests/test_debian_9.py:460]: I'm not saying to group all compose tests, only subsets. For example, take a look at the compose tests for java. The only difference between them are that one looks for a dictionary's value to be "oracle:8" and the other "openjdk:8".
